WiinUPro/Controls/NintyControl.xaml.cs

```diff
@@ -394,6 +394,32 @@ public void LoadProfile(string fileName)
 
             if (loadedProfile != null)
             {
+                if (fileName.Length > 6 & fileName[fileName.Length - 6] == '_')
+                {
+                    int n = (int)Char.GetNumericValue(fileName[fileName.Length - 5]);
+                    bool changeProfile = false;
+                    if (n == 1 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_A))

```

Comment:
You can instead check if n is between (int)Deivce_A and (int)Device_D

Reply:
I'm not sure if I follow. Do you merely wish for 1, 2, 3 & 4 to be replaced with (int)Device_A, (int)Device_B etc. or do you suggest further restructuring of lines 401-416? If so, in what manner?

If you meant checking whether or not the value of n is between 1 ((int)Deivce_A) and 4 ((int)Device_D), then I'm unsure as to what to do with this information, as it doesn't tell me much about whether or not a profile change is due.

On a side note, since all the if-statements have the same result it would be possible to combine all the checks into a single if-statement, although said if-statement would become quite long.

One may alternatively rewrite this section of code as to not rely on recursion.

Reply:
Yeah, you can actually reduce this down quite a bit like so

```suggestion
                string[] fileNameParts = fileName.Split('_');
                string fileNameEnding = fileNameParts[fileNameParts.Length - 1].Replace(".wup", "");
                
                if (int.TryParse(fileNameEnding, out int profileNum)
                    && profileNum >= (int)ScpDirector.XInput_Device.Device_A
                    && profileNum <= (int)ScpDirector.XInput_Device.Device_D
                    && ScpDirector.Access.IsConnected((ScpDirector.XInput_Device)profileNum)
                {
                    // load a different profile
                }
```

Also, what's the behavior if all four controllers are connected? Looks for profile_5.wup and fails if it doesn't exist?

Reply:
Do you prefer using replace or removing the last 4 characters to get rid of the ".wup"?
```C#
ScpDirector.Access.IsConnected((ScpDirector.XInput_Device)profileNum)
```
I didn't know that was possible! Your code snippet above is the behavior I have been looking for which would allow for code simplification here and in MainWindow.xaml.cs.


Due to the possibility of extensions changing the profile the additional check below is also necessary to add to your suggested code, other than that it seems perfect.
```C#
&& !_rumbleSubscriptions[profileNum - 1]
```
_The code above is only one potential way to go about handling extension specific profiles._

Reply:
Remove is good too. I actually forgot about it because I use Replace so frequently.

Ah yes, I saw that it should be added when I was reading it but forgot to add it to the actual example.